django/applications/catmaid/control/cropping.py

```diff
@@ -496,14 +480,16 @@ def extract_substack_no_rotation(job) -> List:
                                      (estimated_total_size,
                                       settings.GENERATED_FILES_MAXIMUM_SIZE))
                 # Draw the image onto result image
-                cropped_slice.composite( image, ip.x_dst, ip.y_dst, co.OverCompositeOp )
+                cropped_slice.paste(image, ip.x_dst, ip.y_dst)
                 # Delete tile image - it's not needed anymore
                 del image
 
-            if cropped_slice:
+            if cropped_slice:  # TODO: always true?
                 # Optionally, use only a single channel
                 if job.single_channel:
-                    cropped_slice.channel( ChannelType.RedChannel )
+                    # r g b
```

Comment:
Will this fail if the slice is already in a single-channel mode?

Reply:
Possibly, but so far I just enforce loading the image patches as RGB data (see `Image` constructor comment). This of course doesn't work with all image data (e.g. N5), but these types aren't supported by the cropping code anyway at the moment. Therefore I think it's okay to keep it like this for now.